index.js

```diff
@@ -18,15 +18,18 @@ module.exports = (hermione, opts) => {
         if (!config.browsers.includes(browserId)) {
             return;
         }
+        async function reassertView(baseAssertViewFn, ...args) {
+            await baseAssertViewFn.apply(this, args);
 
-        const baseAssertView = browser.assertView.bind(browser);
-        browser.addCommand('assertView', async function(...args) {
-            await baseAssertView(...args);
+            const assertViewResults = browser.executionContext.hermioneCtx.assertViewResults.get();
 
-            const assertViewResults = this.executionContext.hermioneCtx.assertViewResults.get();
+            await reassertLastResult(assertViewResults, config, browser.executionContext.ctx.currentTest);
+        }
 
-            await reassertLastResult(assertViewResults, config, this.executionContext.ctx.currentTest);
-        }, true);
+        browser.overwriteCommand('assertView', reassertView);
+        if (browser.__propertiesObject__.assertView) {
```

Comment:
doesn't look like a public API

Reply:
> doesn't look like a public API

In the near future, https://github.com/gemini-testing/hermione/pull/652 will be merged, allowing us to call elem.assertView, and if we skip this check, this plugin won't work in versions below 5.X.0, but this condition improves backward compatibility, so this plugin could work in hermione@4. Is there another way to check if element's prototype has a method?

Reply:
Take a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/prototype
But I think you shouldn't check browser prototype for a specific method at all. It is a part of internal realisation where it stores new method. You should just overwrite command, and may be do some checks inside actual method

Reply:
> You can use `try-catch` block with matching on specific error

Turns out, i can't

```
it('test', async ({browser}) => {
    browser.addCommand('foo', () => 'bar'); // Browser scope
    try {
        await browser.overwriteCommand('foo', () => 'bar', true); // Elem scope
    } catch (e) {
        console.log(e); // Not reached
    }
    await browser.url('https://www.npmjs.com'); // Throws "Error: overwriteCommand: no command to be overwritten: foo"
})
```

This code throws error on `await browser.url(...)`

Any new ideas?

Reply:
Nope for now. Looks like it demands more time for research than a 5 minutes of code review.
In a pinch you can give up the backwards compatibility

Reply:
We agreed to release a major version for this plugin which will include changes:
- migrate to wdio7 (first commit)
- overwrite assertView for element with specifying required version of hermione in peerDeps (next commit)